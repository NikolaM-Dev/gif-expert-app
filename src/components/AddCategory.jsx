import { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (evt) => {
    setInputValue(evt.target.value);
    console.log(evt.target.value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    console.log('handleSubmit', inputValue);

    if (inputValue.trim().length >= 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        onChange={handleInputChange}
        placeholder="Search..."
        type="text"
        value={inputValue}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
