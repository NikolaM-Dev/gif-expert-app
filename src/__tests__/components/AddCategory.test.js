import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import AddCategory from 'components/AddCategory';

describe('Tests on <AddCategory />', () => {
  const setCategories = jest.fn();
  const value = 'Hello World!';
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  it('should show <AddCategory /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the text input', () => {
    const input = wrapper.find('input');

    input.simulate('change', { target: { value } });
  });

  it('should not post the information in the submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  it('should call the getCategories and clean up the text input', () => {
    // 1. Simular el inputChange
    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });
    // 2. Simular el submit
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    // 3. setCategories se debe haber llamado
    expect(setCategories).toHaveBeenCalledTimes(1);
    // 3.1 Con una funcion como parametro
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    // 4. El valor del input debe estar en ''
    expect(input.prop('value')).toBe('');
  });
});
