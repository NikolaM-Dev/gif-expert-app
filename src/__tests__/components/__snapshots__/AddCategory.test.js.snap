// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> should show <AddCategory /> correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
